### ISSUE TITLE
Net-Ping: status in core-p7 branch

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#22]
In this ticket we will track our core-p7 progress on `dist/Net-Ping`.

As of the morning of Jul 08 2020, we are getting this output from `make test` for this distro.

```
t/../dist/Net-Ping/t/000_load ...................................... ok
t/../dist/Net-Ping/t/001_new ....................................... ok
t/../dist/Net-Ping/t/010_pingecho .................................. ok
t/../dist/Net-Ping/t/110_icmp_inst ................................. Bareword found where operator expected at t/110_icmp_inst.t line 24, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/110_icmp_inst.t line 24, near "Net::Ping "icmp""
	(Do you need to predeclare Net::Ping?)
syntax error at t/110_icmp_inst.t line 24, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/110_icmp_inst.t line 25.
Execution of t/110_icmp_inst.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 2 tests, saw 1
t/../dist/Net-Ping/t/120_udp_inst .................................. Bareword found where operator expected at t/120_udp_inst.t line 25, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/120_udp_inst.t line 25, near "Net::Ping "udp""
	(Do you need to predeclare Net::Ping?)
syntax error at t/120_udp_inst.t line 25, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/120_udp_inst.t line 26.
Execution of t/120_udp_inst.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 2 tests, saw 1
t/../dist/Net-Ping/t/130_tcp_inst .................................. Bareword found where operator expected at t/130_tcp_inst.t line 24, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/130_tcp_inst.t line 24, near "Net::Ping "tcp""
	(Do you need to predeclare Net::Ping?)
syntax error at t/130_tcp_inst.t line 24, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/130_tcp_inst.t line 25.
Execution of t/130_tcp_inst.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 2 tests, saw 1
t/../dist/Net-Ping/t/140_stream_inst ............................... Bareword found where operator expected at t/140_stream_inst.t line 24, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/140_stream_inst.t line 24, near "Net::Ping "stream""
	(Do you need to predeclare Net::Ping?)
syntax error at t/140_stream_inst.t line 24, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/140_stream_inst.t line 25.
Execution of t/140_stream_inst.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 2 tests, saw 1
t/../dist/Net-Ping/t/150_syn_inst .................................. Bareword found where operator expected at t/150_syn_inst.t line 25, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/150_syn_inst.t line 25, near "Net::Ping "syn""
	(Do you need to predeclare Net::Ping?)
syntax error at t/150_syn_inst.t line 25, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/150_syn_inst.t line 26.
Execution of t/150_syn_inst.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 2 tests, saw 1
t/../dist/Net-Ping/t/190_alarm ..................................... skipped
t/../dist/Net-Ping/t/200_ping_tcp .................................. skipped
t/../dist/Net-Ping/t/250_ping_hires ................................ skipped
t/../dist/Net-Ping/t/300_ping_stream ............................... skipped
t/../dist/Net-Ping/t/400_ping_syn .................................. skipped
t/../dist/Net-Ping/t/410_syn_host .................................. skipped
t/../dist/Net-Ping/t/420_ping_syn_port ............................. skipped
t/../dist/Net-Ping/t/450_service ................................... Bareword found where operator expected at t/450_service.t line 28, near "new IO::Socket::INET"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/450_service.t line 38, near "new IO::Socket::INET"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/450_service.t line 65, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/450_service.t line 65, near "Net::Ping "tcp""
	(Do you need to predeclare Net::Ping?)
Bareword found where operator expected at t/450_service.t line 102, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/450_service.t line 102, near "Net::Ping "syn""
	(Do you need to predeclare Net::Ping?)
Bareword found where operator expected at t/450_service.t line 119, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/450_service.t line 119, near "Net::Ping "syn""
	(Do you need to predeclare Net::Ping?)
Bareword found where operator expected at t/450_service.t line 145, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/450_service.t line 145, near "Net::Ping "syn""
	(Do you need to predeclare Net::Ping?)
Bareword found where operator expected at t/450_service.t line 163, near "new Net::Ping"
	(Do you need to predeclare new?)
syntax error at t/450_service.t line 29, near "new IO::Socket::INET
  "
Global symbol "$sock1" requires explicit package name (did you forget to declare "my $sock1"?) at t/450_service.t line 34.
syntax error at t/450_service.t line 39, near "new IO::Socket::INET
  "
Global symbol "$sock2" requires explicit package name (did you forget to declare "my $sock2"?) at t/450_service.t line 44.
syntax error at t/450_service.t line 65, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/450_service.t line 67.
syntax error at t/450_service.t line 102, near "new Net::Ping "
syntax error at t/450_service.t line 119, near "new Net::Ping "
syntax error at t/450_service.t line 145, near "new Net::Ping "
syntax error at t/450_service.t line 163, near "new Net::Ping "
t/450_service.t has too many errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 26 tests, saw 1
t/../dist/Net-Ping/t/500_ping_icmp ................................. Bareword found where operator expected at t/500_ping_icmp.t line 58, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/500_ping_icmp.t line 58, near "Net::Ping "icmp""
	(Do you need to predeclare Net::Ping?)
syntax error at t/500_ping_icmp.t line 58, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/500_ping_icmp.t line 59.
Execution of t/500_ping_icmp.t aborted due to compilation errors.
FAILED--no leader found
t/../dist/Net-Ping/t/501_ping_icmpv6 ............................... Bareword found where operator expected at t/501_ping_icmpv6.t line 49, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/501_ping_icmpv6.t line 49, near "Net::Ping "icmpv6""
	(Do you need to predeclare Net::Ping?)
syntax error at t/501_ping_icmpv6.t line 49, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/501_ping_icmpv6.t line 52.
Execution of t/501_ping_icmpv6.t aborted due to compilation errors.
FAILED--no leader found
t/../dist/Net-Ping/t/510_ping_udp .................................. Bareword found where operator expected at t/510_ping_udp.t line 24, near "new Net::Ping"
	(Do you need to predeclare new?)
String found where operator expected at t/510_ping_udp.t line 24, near "Net::Ping "udp""
	(Do you need to predeclare Net::Ping?)
syntax error at t/510_ping_udp.t line 24, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/510_ping_udp.t line 27.
Execution of t/510_ping_udp.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--expected 3 tests, saw 1
t/../dist/Net-Ping/t/520_icmp_ttl .................................. Bareword found where operator expected at t/520_icmp_ttl.t line 23, near "new Net::Ping"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/520_icmp_ttl.t line 27, near "new Net::Ping"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/520_icmp_ttl.t line 31, near "new Net::Ping"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/520_icmp_ttl.t line 39, near "new Net::Ping"
	(Do you need to predeclare new?)
syntax error at t/520_icmp_ttl.t line 23, near "new Net::Ping "
Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at t/520_icmp_ttl.t line 24.
syntax error at t/520_icmp_ttl.t line 27, near "new Net::Ping "
syntax error at t/520_icmp_ttl.t line 31, near "new Net::Ping "
syntax error at t/520_icmp_ttl.t line 39, near "new Net::Ping "
Execution of t/520_icmp_ttl.t aborted due to compilation errors.
# Looks like your test exited with 255 just after 1.
FAILED--non-zero wait status: 65280


```
These results were generated on an unthreaded build on Linux.  The results might differ significantly if run on a threaded build.  There have been subsequent commits to this branch which may have impacted these results.  YMMV.

**Comment by atoomic** [comment]
all issues should have been fixed by 9e5ecda8

**Comment by jkeenan** [comment]
Yes, as of v5.32.0-576-g9e5ecda8ae, Net-Ping is testing clean.  Thanks for your quick action.

> all issues should have been fixed by [9e5ecda](https://github.com/atoomic/perl/commit/9e5ecda8ae6aac420410b9256cde7e3486955f91)

Yes, as of v5.32.0-576-g9e5ecda8ae, Net-Ping is testing clean.  Thanks for your quick action.
